libs/collectives/docs/index.rst

```diff
@@ -15,6 +15,21 @@ The collectives module exposes a set of distributed collective operations. Those
 can be used to exchange data between participating sites in a coordinated way.
 At this point the module exposes the following collective primitives:
 
-* ``all_to_all``: each participating site provides its element of the data to
-  collect while all participating sites receive the data from every other site.
+* :cpp:class:`hpx::collectives::all_reduce`: performs a reduction on data from
+  each participating site.
+* :cpp:class:`hpx::collectives::all_to_all`: each participating site provides its
+  element of the data to collect while all participating sites receive the data
+  from every other site.
+* :cpp:class:`hpx::lcos:barrier`: distributed barrier.
+* :cpp:func:`hpx::lcos::broadcast`: performs a given action on all given global
+  identifiers.
+* :cpp:func:`hpx::lcos::fold`: performs a fold with a given action on all given
+  global identifiers.
+* :cpp:func:`hpx::lcos::gather`: gathers values from all participating sites.
+* :cpp:class:`hpx::lcos::latch`: distributed latch.
+* :cpp:func:`hpx::lcos::reduce`
```

Comment:
@hkaiser what is the fundamental difference between `all_reduce` and `reduce`?

Reply:
@msimberg `reduce` collects the result of an reduction operation at a single 'endpoint', whicl `all_reduce` delivers the reduction result to all involved sites. This is very similar to the `MPI_Reduce` and `MPI_AllReduce` semantics (see for instance https://mpitutorial.com/tutorials/mpi-reduce-and-allreduce/).

Reply:
Ah, of course. Thanks! I'll fill in that part.